### PR TITLE
`feature/npc` --> `quality-assurance`

### PR DESCRIPTION
**Pull request by gmrush**
Branch will remain open as there is plenty more work related to NPCs.

## Fixes:
* `WizardZoneNPC` service options.
* Command parameter parsing.
* Dyes on purchased items not displaying properly.
* Gold balance exceeding gold pouch.

## Changelog:
* Added `WizardZoneNPC`.
* Added `InteractService`.
* Added `ShopService`
* Added NPC interaction.
* Added buy/selling items from/to NPC shops.
* Added individual NPC shop inventories.
* Added player and NPC wizbangs.
* Added character gold persistence.
* Added gold related commands.
* Added `DragonNPCTool`.
* Readded persistent player marks.

### CODE REVIEW

File: src/CoreLib/Game/Services/InventoryService.cs [hunk 1]
```diff
@@ -135,4 +135,25 @@ protected static Props Props(SessionActor parentActor)
...
+
+    [MessageHandler(typeof(WIZARD_12_PROTOCOL.MSG_PLAYERWIZBANG))]
+    private void ReceivePlayerWizbang(WIZARD_12_PROTOCOL.MSG_PLAYERWIZBANG message) {
+        var wizard = GetActiveWizard();
+
+        switch (message.StateName) {
+            case "SpellbookWizbang":
+                ZoneBroadcast(new GAME_5_PROTOCOL.MSG_WIZBANG() {
+                    GameObjectID = wizard.CharId,
+                    WizBangID = StringHash.Compute("Registrar")
```
Comment (Jooty) [inline src/CoreLib/Game/Services/InventoryService.cs:147]: Should do an enum here like we did for `MagicSchool`.

File: src/CoreLib/WizardData/Collections/WizardCollection.cs [hunk 2]
```diff
@@ -123,6 +123,21 @@ public static class WizardCollection {
         session.SaveChanges();
     }
 
+    public static void UpdateCharcterMarkedLocation(Wizard characrer, Vector3 location, Vector3 orientation, string ZoneName) {
```
Comment (Jooty) [inline src/CoreLib/WizardData/Collections/WizardCollection.cs]: method name typo & needs docs

**Review by Jooty — CHANGES_REQUESTED** [review]
good stuff, just a typo and one refactor remaining

**Review by Jooty — COMMENTED** [review]
superb, merge when comfortable

**Review by Jooty — COMMENTED** [review]
godspeed, npc

**Review by Jooty — APPROVED** [review]

**Review by Phill030 — COMMENTED** [review]
🍕

**Pull request merged**